internal/simulation/simulation.go

```diff
@@ -14,52 +19,49 @@ const (
 )
 
 type Simulation struct {
-	done     <-chan struct{}
 	houseImg *ebiten.Image
+	sess     *mcp.Session
+	state    state.State
+}
+
+func New(ctx context.Context, laddr string) (*Simulation, error) {
+	houseImg, err := openImage("./assets/house.png")
+	if err != nil {
+		return nil, err
+	}
 
-	// TODO: generalize to multiplayer
-	inputBuffer <-chan state.Input
+	ln, err := mcp.Listen(laddr, mcp.WithLogger(slog.Default()))
+	if err != nil {
+		return nil, err
+	}
+	sess, err := ln.Accept(ctx)
+	if err != nil {
+		return nil, err
+	}
 
-	state state.State
+	return &Simulation{
+		houseImg: ebiten.NewImageFromImage(houseImg),
+		sess:     sess,
+		state:    state.State{},
+	}, nil
 }
 
-func New(done <-chan struct{}, houseImg image.Image) *Simulation {
-	// bufferred channel for testing
-	// TODO: should be a battle tested jitter buffer
-	ch := make(chan state.Input, 1)
-	// this "mocks" inputs coming in from a single client
-	go func() {
-		defer close(ch)
-
-		t := time.NewTicker(deltaTickTime)
-		defer t.Stop()
-
-		for {
-			select {
-			case <-done:
-				return
-			case <-t.C:
-			}
-
-			input := state.Input{
-				Left:  ebiten.IsKeyPressed(ebiten.KeyH),
-				Down:  ebiten.IsKeyPressed(ebiten.KeyJ),
-				Up:    ebiten.IsKeyPressed(ebiten.KeyK),
-				Right: ebiten.IsKeyPressed(ebiten.KeyL),
-			}
-
-			select {
-			case ch <- input:
-			default:
-			}
-		}
-	}()
+func openImage(name string) (img image.Image, err error) {
```

Comment:
Again, moving `openImage` from cmd/server/main.go to here is not related to this commit.